community/ceph/APKBUILD

```diff
@@ -188,9 +207,21 @@ check() {
 	ctest
 }
 
+# This subpackage is a little weird and a side effect of trying to keep
+# the daemon packages as small as possible. We might be able to roll this into
+# the base/common package later if we can slim those down or break them up in a
+# different way
+user() {
+	pkgdesc="Minimal package to create a user for other ceph packages."
+	depends=""
+	install="$pkgname-user.pre-install"
```

Comment:
Move to the top of `APKBUILD` please.

Reply:
I'm not sure I follow. You want the whole function moved or just the coment?

Reply:
Sorry, the particular line I put this comment on.

Reply:
Does that make it run the pre-install script for every subpackage? Because yeah, then I could just drop the whole `-user` subpackage altogether.

Reply:
It won't change the behavior, but does make it easier to see what is going on.

The part before `.pre-install` is matched against the `pkgname` or `subpkgname` to determine which package contains that script.

Reply:
Thanks for explaining that for me. Not sure I would have ever figured that out on my own. I'll sort this out. Thanks.

For future reference, if I needed different scripts per subpackage, I'd still need to do it this way, right? Or would I put a list of install scripts and abuild would figure it all out?

Reply:
You should put a list at the top and let `abuild` sort it out.

For example, `install="$pkgname.pre-install $pkgname-dev.pre-install"` will put each file into the correct apk archive.